src/runtime/classmanager.cs

```diff
@@ -209,11 +267,16 @@ private static void InitClassBase(Type type, ClassBase impl)
             IntPtr dict = Marshal.ReadIntPtr(tp, TypeOffset.tp_dict);
 
 
+            if (impl.dotNetMembers == null)
+            {
+                impl.dotNetMembers = new List<string>();
+            }
```

Comment:
I am uncertain about thread safety here.

Reply:
I think it is safe. If more than one thread execute this method at the same time, the list will be filled anyway. Duplicate members are not an issue as the code that clears the members deals with `KeyError`s.

Reply:
The thread safety issue is no worse than before; the entry is in GetClass(Type). Two threads that hit `cache.TryGetValue(type, out cb)` simultaneously will each then create a `ClassBase` entry and `Add` it to the cache -- Dictionary isn't thread-safe.